server/src/main/java/org/opensearch/index/mapper/ConstantKeywordFieldMapper.java

```diff
@@ -0,0 +1,178 @@
+            if (!node.containsKey("value")) {
+                throw new OpenSearchParseException("Field [" + name + "] is missing required parameter [value]");
+            }
+            Object value = node.remove("value");
+            if (!(value instanceof String)) {
+                throw new OpenSearchParseException("Field [" + name + "] is expected to be a string value");
+            }
+            return new Builder(name, (String) value);
+        }
+    }
+
+    private static ConstantKeywordFieldMapper toType(FieldMapper in) {
+        return (ConstantKeywordFieldMapper) in;
+    }
+
+    /**
+     * Builder for the binary field mapper
+     *
+     * @opensearch.internal
+     */
+    public static class Builder extends ParametrizedFieldMapper.Builder {
+
+        private final Parameter<String> value;
+
+        public Builder(String name, String value) {
+            super(name);
+            this.value = Parameter.stringParam("value", false, m -> toType(m).value, value);
+        }
+
+        @Override
+        public List<Parameter<?>> getParameters() {
+            return Arrays.asList(value);
+        }
+
+        @Override
+        public ConstantKeywordFieldMapper build(BuilderContext context) {
+            return new ConstantKeywordFieldMapper(
+                name,
+                new ConstantKeywordFieldMapper.ConstantKeywordFieldType(buildFullName(context), value.getValue()),
+                multiFieldsBuilder.build(this, context),
+                copyTo.build(),
+                this
+            );
+        }
+    }
+
+    /**
+     * Field type for Index field mapper
+     *
+     * @opensearch.internal
+     */
+    static final class ConstantKeywordFieldType extends ConstantFieldType {
+
+        protected final String value;
+
+        public ConstantKeywordFieldType(String name, String value) {
+            super(name, Collections.emptyMap());
+            this.value = value;
+        }
+
+        @Override
+        public String typeName() {
+            return CONTENT_TYPE;
+        }
+
+        @Override
+        protected boolean matches(String pattern, boolean caseInsensitive, QueryShardContext context) {
+            return Regex.simpleMatch(pattern, value, caseInsensitive);
+        }
+
+        @Override
+        public Query existsQuery(QueryShardContext context) {
+            return new MatchAllDocsQuery();
+        }
+
+        @Override
+        public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName, Supplier<SearchLookup> searchLookup) {
+            return new ConstantIndexFieldData.Builder(fullyQualifiedIndexName, name(), CoreValuesSourceType.BYTES);
+        }
+
+        @Override
+        public ValueFetcher valueFetcher(QueryShardContext context, SearchLookup searchLookup, String format) {
+            throw new UnsupportedOperationException("Cannot fetch values for internal field [" + name() + "].");
+        }
```

Comment:
I see that we disallow value-fetching for internal fields, but I think we can (and should?) support it for constant keyword fields. 

In particular, I think we can return `(l) -> Collections.singletonList(value)`.

Reply:
@msfroh 
Yes makes Sense
So I implemented `valueFetcher` but I am unable to figure out when is valueFetcher called.
I want to know how it is used to write tests for this